bitcoinology/what-is-lightning-network.md

```diff
@@ -0,0 +1,90 @@
ge. It was written in a kind of strange way. I don't know how many people... Hands up if you have actually read the Lightning Network paper. I mean, more than like a few lines. More than a few lines. A few lines doesn't count. All right? There's a few. Well, well done. You're obviously very masochistic people because that paper is not an easy read. And this illustrates, I think, perfectly. This die is one of the final diagrams. It just shows you how complicated the structure of what they were describing was. So it was very difficult to understand, and it was based around a couple of little hacks because actually Bitcoin's blockchain intrinsically didn't support it. Now before I run on, what on earth am I talking about, the Lightning Network paper? What is the Lightning Network, just for the people who don't know what it is? It's basically the idea that given everything I've said in the last 10 minutes, we need a system where we can transfer Bitcoin without actually putting it on the Bitcoin blockchain. Because the Bitcoin blockchain is intrinsically pretty expensive to use. Intrinsically, once enough people start using it. So their idea was, let's make transactions of Bitcoin back and forth, not just back and forth between two people, but between anyone in the world and me, and have it not actually touch the blockchain until some later time. And we'll go into that later, how that works. So that was their idea, but to actually make it work was this incredibly complicated Byzantine structure. And part of what made it complicated was something called malleability. So malleability is just a fancy word for change, really. If something's changeable, then it's malleable. And the problem is that intrinsically at that time, transactions on Bitcoin's blockchain were in some sense malleable in that when you created them, you could then create another version of them that was still valid before it got committed. So I know I'm getting a bit technical there, but basically there was a problem with the Bitcoin's blockchain that made it too changeable. And a solution came out called SegWit, but SegWit got all mixed up with something called the block size war. So while in 2016 these guys were coming up with this clever new scheme and some other people were getting excited about this clever new scheme, even though it was pretty complicated, they were still excited because we could get these transactions off the blockchain. A lot of other people were still arguing about, no, no, no, let's make the block sizes bigger because that way we can get more and more transactions on the blockchain. Okay, so the argument was about should we go off-chain or should we make the blockchain bigger to support more and more transactions. So Jonathan Beer here wrote a book recently called the Block Size War, I think about last year sometime. And it's quite interesting, if you don't know the history of the Block Size War, it is kind of interesting, but that's not the topic of our talk.
```

Comment:
```suggestion
what we're gonna see in the next couple of years is in the background, something else was happening. So this is January 2016, roughly. We have these two guys, Tadge Dryja and Joseph Poon, writing a paper called The Lightning Network Paper. Initially it was something just in the back quarters because it was just technical people, a few developers here and there and so on, researchers who were even remotely interested in this. Because actually this paper was kind of strange. It was written in a kind of strange way. I don't know how many people... Hands up if you have actually read the Lightning Network paper. I mean, more than like a few lines. More than a few lines. A few lines doesn't count. All right? There's a few. Well, well done. You're obviously very masochistic people because that paper is not an easy read. And this illustrates, I think, perfectly. This die is one of the final diagrams. It just shows you how complicated the structure of what they were describing was. So it was very difficult to understand, and it was based around a couple of little hacks because actually Bitcoin's blockchain intrinsically didn't support it. Now before I run on, what on earth am I talking about, the Lightning Network paper? What is the Lightning Network, just for the people who don't know what it is? It's basically the idea that given everything I've said in the last 10 minutes, we need a system where we can transfer Bitcoin without actually putting it on the Bitcoin blockchain. Because the Bitcoin blockchain is intrinsically pretty expensive to use. Intrinsically, once enough people start using it. So their idea was, let's make transactions of Bitcoin back and forth, not just back and forth between two people, but between anyone in the world and me, and have it not actually touch the blockchain until some later time. And we'll go into that later, how that works. So that was their idea, but to actually make it work was this incredibly complicated Byzantine structure. And part of what made it complicated was something called malleability. So malleability is just a fancy word for change, really. If something's changeable, then it's malleable. And the problem is that intrinsically at that time, transactions on Bitcoin's blockchain were in some sense malleable in that when you created them, you could then create another version of them that was still valid before it got committed. So I know I'm getting a bit technical there, but basically there was a problem with the Bitcoin's blockchain that made it too changeable. And a solution came out called SegWit, but SegWit got all mixed up with something called the block size war. So while in 2016 these guys were coming up with this clever new scheme and some other people were getting excited about this clever new scheme, even though it was pretty complicated, they were still excited because we could get these transactions off the blockchain. A lot of other people were still arguing about, no, no, no, let's make the block sizes bigger because that way we can get more and more transactions on the blockchain. Okay, so the argument was about should we go off-chain or should we make the blockchain bigger to support more and more transactions. So Jonathan Beer here wrote a book recently called the Block Size War, I think about last year sometime. And it's quite interesting, if you don't know the history of the Block Size War, it is kind of interesting, but that's not the topic of our talk.
```